Modules/Core/TestKernel/src/itkTestDriverIncludeRequiredFactories.cxx

```diff
@@ -70,6 +100,34 @@ RegisterRequiredFactories()
   itk::ObjectFactoryBase::RegisterFactory(itk::VTKPolyDataMeshIOFactory::New());
 }
 
+void
+RegisterRequiredFFTFactories()
+{
+#if defined(ITK_USE_FFTWD) || defined(ITK_USE_FFTWF)
```

Comment:
Can we avoid duplication with https://github.com/InsightSoftwareConsortium/ITK/blob/test-kernel-register-fft-factories/Modules/Core/TestKernel/include/itkTestDriverIncludeBuiltInFactories.h?

Reply:
`itkTestDriverIncludeRequiredFactories` is referenced in the [`CreateTestDriver`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CMake/ITKModuleTest.cmake#L59) macro while `itkTestDriverIncludeBuildInFactories` is referenced in the [`CreateTestDriver_SupportBuildInFactories`](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CMake/ITKModuleTest.cmake#L113) macro. It looks like `BuildInFactories` was split out from `RequiredFactories` in [this commit](https://github.com/InsightSoftwareConsortium/ITK/commit/a2c7a16e45a35f3c3988b1dac4d349b0a754fc1f) 10 years ago to aid regression testing.

I'm not sure that I fully understand the implications of how `BuildInFactories` is being used here, it looks like it is only invoked in the `Nonunit/Review` and `Nonunit/IntegratedTest` modules. @dzenanz do you maybe know why these modules are using a slightly different testing structure?

If this is an old/outdated structure we can look into consolidating, but the scope of this PR should probably stay limited to adding default factories.

Reply:
I don't know why `Nonunit` tests have a different structure. Maybe a relic of old, maybe because they need it. They do testing which involves multiple modules, so they are different from unit tests.

> we can look into consolidating

We should note this as an issue, unless you are going to do this soon.

Reply:
> They do testing which involves multiple modules, so they are different from unit tests.

Ah okay, good to know.

> We should note this as an issue

Agreed, I will enter an issue to track this discussion separately.

Reply:
See #3004